### ISSUE TITLE
'experimart.interoperability.torch.step' not found

### ISSUE THREAD

**Issue by BestVIncent** [DIAGNijmegen/pathology-hooknet#17]
Hello,
Thank you for your excellent work on hooknet! 
I am very interested in your work. Recently, I noticed that you have made some changes about torch training. This is very exciting, as I am not familiar with tensorflow. But I can't find the 'experimart.interoperability.torch.step' library you mentioned. Did I missing something?
Thanks for your help
Vincent

**Comment by martvanrijthoven** [comment]
Dear Vincent,

Thank you for your kind words about hooknet!

I have been experimenting with a torch implementation of hooknet, which has some differences compared to the tensorflow version. The main differences are:

|              | Torch | Keras |
|--------------|------:|------:|
| Number of branches | 3 | 2 |
| Residual connections | Yes | No |
| Multi-loss | No | Yes |

Also, there are some other minor changes that you can refer to in the code or ask me if anything needs to be clarified. Please note that this torch implementation is still experimental and is not tested or benchmarked extensively. 

I created the "experimart" package as a private library to quickly test things. It is not in a state to make it public and may not be helpful for anyone else, but I will see how that code evolves. For now, I have created a notebook with an example of training for the torch model without needing the experimart package: [pytorch example notebook](https://github.com/DIAGNijmegen/pathology-hooknet/blob/master/notebooks/HookNetPracticalGuideTorch.ipynb)

Just so you know, because the model has three branches, it may be slower than the tensorflow version (both the model and patch extraction). Use multiple CPUs to speed up batch extraction with the [wholeslideimage package](https://github.com/DIAGNijmegen/pathology-whole-slide-data/) if possible. I plan to optimize training speed with multiple GPUs but have yet to look into it.
Even though hooknet-pytorch is still in development, I hope the notebook can still be helpful to you. Please let me know if you have any questions.

Best wishes,
Mart

**Comment by BestVIncent** [comment]
Thank you very much! Your example notebook is very helpful. By the way, is the pre-trained model of hooknet on breast publicly available? If so, how can I obtain it?

**Comment by martvanrijthoven** [comment]
Dear Vincent, 

You are welcome! The pretrained breast model is available on grand-challenge: https://grand-challenge.org/algorithms/hooknet-breast/

Best wishes,
Mart
